judo-runtime-core-dao-core/src/main/java/hu/blackbelt/judo/runtime/core/dao/core/processors/UpdatePayloadDaoProcessor.java

```diff
@@ -223,6 +221,25 @@ void collectStaments(EClass transferObjectType,
         }
 
         references.stream().forEach(currentReference -> {
+                if (updatePayload.containsKey(currentReference.getName())
+                        && updatePayload.get(currentReference.getName()) != null
+                        && getAsmUtils().getMappedReference(currentReference).isPresent()
+                        && getAsmUtils().getMappedReference(currentReference).orElseThrow().isContainment()) {
+                    Collection<Payload> currentPayload;
+
+                    if (isCollection.test(currentReference)) {
+                        currentPayload = updatePayload.getAsCollectionPayload(currentReference.getName());
+                    } else {
+                        currentPayload = ImmutableList.of(updatePayload.getAsPayload(currentReference.getName()));
+                    }
+
+                    if (currentPayload != null && originalPayload.get(getIdentifierProvider().getName()) == null && updatePayload.get(getIdentifierProvider().getName()) != null) {
+                        updatePayload.remove(getIdentifierProvider().getName());
+                        updatePayload.remove(ENTITY_TYPE_KEY);
+                        updatePayload.remove(VERSION);
```

Comment:
Doesn't this cause optimistic locking to fail? @robertcsakany

Reply:
The "if" condition is true if the instance that is intended to be added to the composition already has an ID but the instance itself was missing from the original payload. In that case a new instance must be created (deep copy) and to achieve that, the identifier must removed, making the version entry (as far as I understand) no longer needed.